### PR TITLE
chore(dep): update jmx exporter

### LINKED ISSUE #7398: Update JMX Exporter to 0.17.2 to patch snakeyaml CVEs
JMX Exporter 0.17.0 [shades SnakeYaml 1.30](https://github.com/prometheus/jmx_exporter/blob/main/jmx_prometheus_javaagent/pom.xml#L55).  Strimzi shadowed snakeyaml to 1.32, but because this one is shaded, it can't be overridden.  

JMX Exporter 0.17.2 has been rebuilt with SnakeYaml 1.32.   Update Strimzi to this version to patch the vulnerabilities in earlier versions of snakeyaml.

### PR DESCRIPTION
**Pull request by Boojapho**
Signed-off-by: Michael McLeroy <michaelmcleroy@cloudfitsoftware.com>

### Type of change

- Enhancement / new feature 
- Closes #7398 

### Description

Updates JMX Exporter version to 0.17.2

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: docker-images/artifacts/kafka-thirdparty-libs/3.1.x/pom.xml [hunk 1]
```diff
@@ -23,7 +23,7 @@
         <kafka-mirror-maker-2-extensions.version>1.2.0</kafka-mirror-maker-2-extensions.version>
         <kafka-kubernetes-config-provider.version>1.0.1</kafka-kubernetes-config-provider.version>
         <kafka-env-var-config-provider.version>1.0.0</kafka-env-var-config-provider.version>
-        <jmx-prometheus-javaagent.version>0.17.0</jmx-prometheus-javaagent.version>
+        <jmx-prometheus-javaagent.version>0.17.2s</jmx-prometheus-javaagent.version>
```
Comment (scholzj) [inline docker-images/artifacts/kafka-thirdparty-libs/3.1.x/pom.xml:26]: I guess `0.17.2s` is a typo and it should be `0.17.2`?

Comment (Boojapho) [reply docker-images/artifacts/kafka-thirdparty-libs/3.1.x/pom.xml]: Yes.  Fixed

**Review by scholzj — APPROVED** [review]
Thanks for the PR.

**Review by ppatierno — APPROVED** [review]

**Comment by scholzj** [comment]
Thanks for the PR @Boojapho

**Pull request merged**